### PR TITLE
Re-implemented murmur2 partitioner, to be consistent with Kafka Java client

### PR DESCRIPTION
**Pull request by vuza**
Using https://github.com/vuza/murmur2-partitioner for murmur2 partitioning, which is tested to be consistent with Kafka's Java client. Also fixing this two issues: https://github.com/nodefluent/node-sinek/issues/40, https://github.com/nodefluent/node-sinek/issues/39

### CODE REVIEW

**Comment by vuza** [comment]
Close https://github.com/nodefluent/node-sinek/issues/40 and https://github.com/nodefluent/node-sinek/issues/39 after merge!

File: lib/librdkafka/NProducer.js [hunk 1]
```diff
@@ -272,7 +259,16 @@ class NProducer extends EventEmitter {
...
-    return this._murmur(key) % partitionCount;
+    switch (this._murmurHashVersion) {
+    case "2":
+      return murmur2Partitioner.partition(key, partitionCount);
+        
+    case "3":
+      return murmur.v3(key) % partitionCount;
+        
+    default:
+      throw new Error(`${this._murmurHashVersion} is not a supported murmur hash version. Choose '2' or '3'.`);
```
Comment (krystianity) [inline lib/librdkafka/NProducer.js]: this throws on every message you are trying to produce, evaluating this in the constructor only throws once and kills the producer. I personally think that would be a better dev-experience.

**Review by krystianity — CHANGES_REQUESTED** [review]

**Comment by vuza** [comment]
You are right! I fixed that!

**Comment by krystianity** [comment]
I still dont like pulling two modules in just for the partitioner..however lets get this released already..

File: lib/librdkafka/NProducer.js [hunk 2]
```diff
@@ -272,7 +272,7 @@ class NProducer extends EventEmitter {
       throw new Error("partitionCount must be number.");
     }
 
-    return this._murmur(key) % partitionCount;
+    this._murmur(key, partitionCount);
```
Comment (eli5zen) [inline lib/librdkafka/NProducer.js:275]: I believe a little bug was introduced in this release - previously the function returned the murmur result, but now it only calls the murmur function and always returns `undefined`. As a result, manual partition key cannot be provided by the application and a random partition is selected.

Comment (krystianity) [reply lib/librdkafka/NProducer.js:275]: @elizehavi you are right, good find! Bad review on my part back then. I have fixed the issue and released sinek@6.22.2
